src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -517,3 +517,140 @@ bool java_bytecode_convert_class(
 
   return true;
 }
+
+/// For a given generic type parameter, check if there is a parameter in the
+/// given vector of replacement parameters with a matching name. If yes,
+/// replace the identifier of the parameter at hand by the identifier of
+/// the matching parameter.
+/// Example: if \p parameter has identifier `java::Outer$Inner::T` and
+/// there is a replacement parameter with identifier `java::Outer::T`, the
+/// identifier of \p parameter gets set to `java::Outer::T`.
+/// \param parameter
+/// \param replacement_parameters
+void find_and_replace_parameter(
+  java_generic_parametert &parameter,
+  const std::vector<java_generic_parametert> &replacement_parameters)
+{
+  // get the name of the parameter, e.g., `T` from `java::Class::T`
+  std::string parameter_full_name =
+    as_string(parameter.type_variable_ref().get_identifier());
+  std::string parameter_name =
+    parameter_full_name.substr(parameter_full_name.rfind(':') + 1);
+
+  // check if there is a replacement parameter with the same name
+  const auto replacement_parameter_p = std::find_if(
+    replacement_parameters.begin(),
+    replacement_parameters.end(),
+    [&parameter_name](const java_generic_parametert &replacement_param)
+    {
+      std::string replacement_parameter_full_name =
+        as_string(replacement_param.type_variable().get_identifier());
+      std::string replacement_parameter_name =
+        replacement_parameter_full_name.substr(
+          replacement_parameter_full_name.rfind(':') + 1);
+      return parameter_name == replacement_parameter_name;
+    });
```

Comment:
if I understand correctly, then this will test on the generic parameter name only? Is then the hypothesis that `replacement_parameters` will only contain viable candidates? If so, please add this in the documentation.

Reply:
Added to documentation and also added a precondition.